### ISSUE TITLE
Should "emit -points" stamp points with time if none is present?

### ISSUE THREAD

**Issue by go-oleg** [juttle/juttle#77]
Currently if you generate some timeless points with `emit -points`, time is added to them.

juttle:
```raw
emit -points [ { a: 1 }, { a: 2 } ]
```

output:
```raw
┌────────────────────────────────────┬──────────┐
│ time                               │ a        │
├────────────────────────────────────┼──────────┤
│ 2015-12-29T03:52:42.593Z           │ 1        │
├────────────────────────────────────┼──────────┤
│ 2015-12-29T03:52:43.593Z           │ 2        │
└────────────────────────────────────┴──────────┘
```

I don't think we want to be mutating points the user specified.

Thoughts?

**Comment by dmehra** [comment]
@dmajda please weigh in on this. The behavior of adding timestamps to -points array was surprising to me when I was playing around with `join` like a new user might. When I thought about it, it made sense since `emit` always generates points with time... and yet, the initial reaction was consternation. 

If we want to keep the behavior, let's make the docs very clear. The current emit doc says 
```
-points	An array of points to stream
```
with no mention of adding time to them.

**Comment by demmer** [comment]
It seems to me the default should be to emit timeless points, but if the user specifies a `-every` option, then emit should timestamp accordinvgly.

**Comment by go-oleg** [comment]
`-every` with `-points` could also mean emit all of these points every `-every` seconds instead of one of these points every `-every` seconds serially. Not sure which is more common/intuitive, just bringing it up.

**Comment by dmajda** [comment]
I agree the behavior is unintuitive. I think its rationale was that `-points` was mostly used in tests (it replaced old `demo ticker` there) where the auto-add behavior helped with creating meaningful point sequences.

I tried to comment-out the relevant code and only 8 test failures popped-up, 1 of which was in `emit` tests. So we don’t depend on the behavior much and I’d vote for changing it. As long as we support timeless points (which I’m not a fan of at all, as some may remember from old discussions), let’s just emit what is in `-points` verbatim (but still disallow a mix of timestamped and non-timestamped points).

> It seems to me the default should be to emit timeless points, but if the user specifies a -every option, then emit should timestamp accordinvgly.

This is little bit too magical. I’d say `-every` and `-points` should be mutually exclusive. And if we really have a strong use case for time auto-add, let’s invent a dedicated option to enable that.

**Comment by demmer** [comment]
@dmajda That sounds good to me.

**Comment by dmajda** [comment]
OK, morphing into a task.

**Comment by go-oleg** [comment]
Judging by the issue @dmehra is seeing in her followup comment in https://github.com/juttle/juttle/issues/76#issuecomment-174596640, we should also not emit ticks and if possible, emit all points at once instead of one every second (though the scheduler will probably be in charge of this). From my understanding, whether we emit all the points at the same time or not shouldn't have an effect on the results of any downstream `merge`s and `join`s since they will wait for an EOF from this stream before proceeding (because all points are timeless).

**Comment by dmajda** [comment]
Agreed with not emitting ticks and emitting all points at once.

@bkutil Could you maybe look at implementing this (if it makes sense before finishing the scheduler work)?
